tests/test_analysis.py

```diff
@@ -64,11 +65,17 @@ def test_can_skip_dot_folder(self):
         scanned_names = [os.path.basename(source_path) for source_path, _ in scanner.source_paths()]
         assert scanned_names == [name_to_include]
 
-    def test_fails_on_non_repo_url(self):
+    def test_succeeds_on_not_git_extension(self):
         non_repo_urls = [["https://github.com/roskakori/pygount/"], ["git@github.com:roskakori/pygount"]]
         for non_repo_url in non_repo_urls:
-            with analysis.SourceScanner(non_repo_url) as scanner, pytest.raises(PygountError):
-                next(scanner.source_paths())
+            with analysis.SourceScanner(non_repo_url) as scanner:
+                _ = list(scanner.source_paths())
+
+    def test_fails_on_non_git_urls(self):
+        non_repo_urls = [["https://no/git/url"], ["https://google.com/nogit"]]
+        for non_repo_url in non_repo_urls:
+            with analysis.SourceScanner(non_repo_url) as scanner, pytest.raises(pygount.Error):
```

Comment:
Use `pytest.raises(pygout.Error, match="...")` to also check the error message. You can use a regex here.

If the expected error message is long, I typically only check the first part of the sentence.